### PR TITLE
[CI] Set up ESLint CI

### LINKED ISSUE #46: ESLint CI

### PR DESCRIPTION
**Pull request by rmanaem**
- Closes #46 

## Checklist

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see https://neurobagel.org/contributing/pull_requests for more info)_
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [x] Tests pass
- [x] Checks pass

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: npm install
```
Comment (surchs) [inline .github/workflows/lint.yaml]: https://docs.npmjs.com/cli/v8/commands/npm-ci might be safer here too. so we don't have accidental dep upgrades that break things in odd ways compared to local

File: src/components/ResultCard.tsx
Comment (surchs) [inline src/components/ResultCard.tsx:1]: guessing this was just a prettier run?

**Review by surchs — APPROVED** [review]
:tada: thanks @rmanaem! one suggestion for `npm ci` - otherwise looks great

:cook:

File: src/components/ResultCard.tsx
Comment (rmanaem) [reply src/components/ResultCard.tsx:1]: Yea

**Comment by surchs** [comment]
:rocket: PR was released in [`v0.1.0`](https://github.com/neurobagel/query-tool/releases/tag/v0.1.0) :rocket:

**Pull request merged**